src/Features/Core/Portable/Completion/Providers/AbstractKeywordCompletionProvider.cs

```diff
@@ -48,23 +49,26 @@ public int GetHashCode(RecommendedKeyword obj)
 
             using (Logger.LogBlock(FunctionId.Completion_KeywordCompletionProvider_GetItemsWorker, cancellationToken))
             {
-                var keywords = await document.GetUnionItemsFromDocumentAndLinkedDocumentsAsync(
+                var completionItems = await document.GetUnionItemsFromDocumentAndLinkedDocumentsAsync(
                     s_comparer,
-                    (doc, ct) => RecommendKeywordsAsync(doc, position, ct),
+                    async (doc, ct) =>
+                    {
+                        var syntaxContext = await CreateContextAsync(doc, position, ct).ConfigureAwait(false);
```

Comment:
The IDE prefers to avoid async lambdas.  can you make this a normal Task returning lambda, and have it call the async helper?